libs/blocks/section-metadata/sticky-section.js

```diff
@@ -17,7 +18,12 @@ function promoIntersectObserve(el, stickySectionEl, options = {}) {
       const isPromoStart = entry.target === stickySectionEl;
       const abovePromoStart = (isPromoStart && entry.isIntersecting)
         || stickySectionEl?.getBoundingClientRect().y > 0;
-      if (entry.isIntersecting || abovePromoStart) el.classList.add('hide-sticky-section');
+
+      if (entry.target === document.querySelector('footer')) {
+        isFooterStart = entry.isIntersecting;
+      }
+
+      if (entry.isIntersecting || abovePromoStart || isFooterStart) el.classList.add('hide-sticky-section');
```

Comment:
I think we can reduce the complexity for this, and not cache `isFooterStart`:
```suggestion
if (entry.isIntersecting || 
    abovePromoStart || 
    (entry.target === document.querySelector('footer') && entry.isIntersecting)) {
  el.classList.add('hide-sticky-section');
}
```